src/etc/inc/system.inc

```diff
@@ -1690,6 +1690,38 @@ function system_ntp_setup_pps($serialport) {
 	return true;
 }
 
+function system_ntp_setup_acts($serialport) {
+	global $config;
+	$acts_device = '/dev/acts0';
+	$serialport = '/dev/'.$serialport;
+
+	if (!file_exists($serialport)) {
+		return false;
+	}
+
+	// Create symlink that ntpd requires
+	unlink_if_exists($acts_device);
+	@symlink($serialport, $acts_device);
+
+	$actsbaud = $config['ntpd']['acts']['speed'];
+	/* Configure the serial port for raw IO and set the speed */
+	mwexec("stty -f {$serialport} raw speed {$actsbaud}");
+	mwexec("stty -f {$serialport}.init raw speed {$actsbaud}");
+
+	/* Remove old /etc/remote entry if it exists */
+	$acts_shortname = "acts0";
+	if (intval(`/usr/bin/grep -c "^{$acts_shortname}" /etc/remote`) != 0) {
+		mwexec("/usr/bin/sed -i '' '-n '/'{$acts_shortname}'/!p' /etc/remote");
+	}
+
+	/* Add /etc/remote entry in case we need to read from the ACTS with tip */
+	if (intval(`/usr/bin/grep -c "^{$acts_shortname}" /etc/remote`) == 0) {
```

Comment:
Rather than running the grep a second time, this should just be an else clause to the if above, since if it fails the `!= 0` test then it must be 0.